firefox_ui_tests/functional/locationbar/test_escape_autocomplete.py

```diff
@@ -0,0 +1,72 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from firefox_ui_harness.decorators import skip_under_xvfb
+from firefox_ui_harness.testcase import FirefoxTestCase
+
+
+class TestEscapeAutocomplete(FirefoxTestCase):
+
+    # setUp copied from test_back_forward
+    def setUp(self):
+        FirefoxTestCase.setUp(self)
+
+        with self.marionette.using_context("content"):
+            self.marionette.navigate("about:blank")
+
+# 1. Clear complete history so there's no interference from previous entries.
+        self.marionette.execute_script("""
+            let count = gBrowser.sessionHistory.count;
+            gBrowser.sessionHistory.PurgeHistory(count);
+        """)
+
+# 2. Open some local pages to set up the test environment.
+        self.test_urls = [
+            'layout/mozilla.html',
+            'layout/mozilla_mission.html',
+            'layout/mozilla_grants.html',
+        ]
+        self.test_urls = [self.marionette.absolute_url(t)
+                          for t in self.test_urls]
+
+        with self.marionette.using_context('content'):
+            for url in self.test_urls:
+                self.marionette.navigate(url)
+            self.assertEquals(self.marionette.get_url(), self.test_urls[-1])
```

Comment:
I wonder if Marionette has an internal check that the requested page has been loaded. Can you please check that?